proposals/0111-linear-types.rst

```diff
@@ -2191,6 +2445,36 @@ would define
 
   type CompatibleWithOne p = 1 ⩽ p
 
+Let bindings and polymorphism
+-----------------------------
+
+This proposal specifies (in particular in `Non-variable linear
+patterns are monomorphic`_) that
+
+1. ``-XLinearTypes`` implies ``-XMonoLocalBinds``.
+2. Polymorphic non-multiplicity-annotated let-bound non-variable patterns are inferred to be unrestricted
+3. Multiplicity annotated let-bound non-variable patterns are not generalised.
+
+MonoLocalBinds
+~~~~~~~~~~~~~~
+
+We could choose not to imply ``-XMonoLocalBinds``. The whole point of
+having ``-XMonoLocalBinds`` is to make (2) irrelevant though. Because
+it feels rather unpredictable. Because ``-XMonoLocalBinds`` is quite
+standard already, being more or less necessary for ``-XGADTs`` and
+``-XTypeFamilies``, so it feels unnecessary to add extra worries by
+not not implying ``-XMonoLocalBinds``.
+
+%Many annotated patterns
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+We could try to generalise multiplicity annotated bindings when they
+are annotated with ``%Many``. But it invites a lot of complication
+(what if it's a type synonym that is equivalent to ``Many``? What if
+it's a type family?) that are hard to justify. In particular since we
+take ``-XMonoLocalBinds`` for granted, where generalisation doesn't
```

Comment:
Again, this last point isn't accurate, but it doesn't matter. I'm fine with `let %Many` disabling generalization. If you want it, just use a type signature!